src/parser/hevc_parser.cpp

```diff
@@ -22,49 +22,18 @@ THE SOFTWARE.
 
 #include "hevc_parser.h"
 
-template <typename T>
-inline T *AllocStruct(const int max_cnt) {
-    T *p = nullptr;
-    try {
-        p = (max_cnt == 1) ? new T : new T [max_cnt];
-    }
-    catch(const std::exception& e) {
-        ERR(STR("Failed to alloc HEVC header struct Data, ") + STR(e.what()))
-    }
-    memset(p, 0, sizeof(T) * max_cnt);
-    return p;
-}
-
 HevcVideoParser::HevcVideoParser() {
     first_pic_after_eos_nal_unit_ = 0;
     m_active_vps_id_ = -1; 
     m_active_sps_id_ = -1;
     m_active_pps_id_ = -1;
```

Comment:
We init the element of the arrays before parsing it. We will need to think about is_received flag init. I'll experiment. But we need to verify the time cut on multiple systems as well (especially on MI300).

Reply:
I suggest initializing absolutely must members on constructor still. is_received has to be initialized to 0, otherwise we might get issues when parsing from the middle of picture etc

Reply:
Agreed. Added in. Now the constructor time increased to from ~50 us to ~180 us, which is ok.